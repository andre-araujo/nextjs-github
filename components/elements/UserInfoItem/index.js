import {
    oneOfType,
    number,
    string,
} from 'prop-types';

import {
    Wrapper,
    Title,
    Description,
} from './styles';

const UserInfoItem = ({
    title,
    description,
    url,
}) => (
    <Wrapper>
        <Title>{title}</Title>
        <Description>
            {
                url &&
                <a href={url} target="_blank">{ description }</a>
            }

            {
                !url && (description || 'Unavaliable')
            }
        </Description>
    </Wrapper>
);

UserInfoItem.defaultProps = {
    title: null,
    description: null,
    url: null,
};

UserInfoItem.propTypes = {
    title: string.isRequired,
    description: oneOfType([
        number,
        string,
    ]),
    url: string,
};

export default UserInfoItem;
