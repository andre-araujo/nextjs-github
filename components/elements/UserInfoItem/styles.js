import styled from 'styled-components';

export const Wrapper = styled.li`
    margin: 1rem 0;
`;

export const Title = styled.h3`
    font-family: AvantGardeDemi;
    font-size: 1rem;
    margin-bottom: .5em;
`;


export const Description = styled.p`
    font-size: .8rem;
    margin-right: .2rem;
    font-family: AvantGardeLight;
`;
