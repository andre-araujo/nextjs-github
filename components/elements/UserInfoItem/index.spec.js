import React from 'react';
import { shallow } from 'enzyme';

import UserInfoItem from './index';

describe('UserInfoItem component', () => {
    it('should match snapshot with Unavaliable description', () => {
        expect(shallow(
            <UserInfoItem
                title="title"
            />,
        )).toMatchSnapshot();
    });

    it('should match snapshot with description', () => {
        expect(shallow(
            <UserInfoItem
                title="title"
                description="description"
            />,
        )).toMatchSnapshot();
    });

    it('should match snapshot with link on description', () => {
        expect(shallow(
            <UserInfoItem
                title="title"
                description="description"
                url="some_link"
            />,
        )).toMatchSnapshot();
    });
});
