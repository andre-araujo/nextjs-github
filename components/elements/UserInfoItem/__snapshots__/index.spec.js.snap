// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoItem component should match snapshot with Unavaliable description 1`] = `
<styled.li>
  <styled.h3>
    title
  </styled.h3>
  <styled.p>
    Unavaliable
  </styled.p>
</styled.li>
`;

exports[`UserInfoItem component should match snapshot with description 1`] = `
<styled.li>
  <styled.h3>
    title
  </styled.h3>
  <styled.p>
    description
  </styled.p>
</styled.li>
`;

exports[`UserInfoItem component should match snapshot with link on description 1`] = `
<styled.li>
  <styled.h3>
    title
  </styled.h3>
  <styled.p>
    <a
      href="some_link"
      target="_blank"
    >
      description
    </a>
  </styled.p>
</styled.li>
`;
